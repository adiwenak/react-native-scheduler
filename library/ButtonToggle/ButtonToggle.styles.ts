import { StyleSheet, TextStyle, ViewStyle } from "react-native"
import { Colours } from "../shared"

interface Style {
  buttonSelected: ViewStyle
  buttonNormal: ViewStyle
  titleNormal: TextStyle
  titleSelected: TextStyle
}

export const styles = StyleSheet.create<Style>({
  buttonNormal: {
    borderRadius: 5,
    backgroundColor: Colours.Grey200,
    justifyContent: "center",
    alignItems: "center",
    marginHorizontal: 2
  },
  buttonSelected: {

  },
  titleNormal: {
    fontSize: 12,
    color: Colours.BlueGrey700,
    marginHorizontal: 7,
    marginVertical: 5
  },
  titleSelected: {
    color: Colours.Grey50
  }
})
