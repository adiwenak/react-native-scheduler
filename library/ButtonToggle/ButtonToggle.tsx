import * as React from "react"
import { Text, TouchableOpacity } from "react-native"
import { styles } from "./ButtonToggle.styles"

interface Props {
  selected: boolean
  title: string
  selectedColour: string
  onPress: () => void
}

interface State {
  selected: boolean
}

export class ButtonToggle extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      selected: props.selected
    }
  }

  public render() {
    const buttonStyle = this.props.selected ?
      [styles.buttonNormal, { backgroundColor: this.props.selectedColour }] : [styles.buttonNormal]
    const titleStyle = this.props.selected ?
      [styles.titleNormal, styles.titleSelected] : [styles.titleNormal]

    return (
      <TouchableOpacity
        testID={"Button-Toggle"}
        style={buttonStyle}
        onPress={this.buttonPress}>
        <Text
          style={titleStyle}
        >
          {this.props.title}
        </Text>
      </TouchableOpacity>
    )
  }

  private buttonPress = () => {
    this.setState({ selected: !this.state.selected })
    this.props.onPress()
  }
}
