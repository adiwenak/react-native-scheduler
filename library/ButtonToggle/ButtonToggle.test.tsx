import { shallow } from "enzyme"
import * as React from "react"
import { create } from "react-test-renderer"
import { ButtonToggle } from "./ButtonToggle"

describe("<ScheduleView />", () => {
  const props = {
    selected: true,
    title: "button1",
    selectedColour: "blue",
    onPress: jest.fn()
  }

  describe("render()", () => {

    it("should render with selected state", () => {
      const component = create(
        <ButtonToggle
          {...props}
        />
      ).toJSON()

      expect(component).toMatchSnapshot()
    })

    it("should render with unselected state", () => {
      const unselected = { ...props, selected: false }
      const component = create(
        <ButtonToggle
          {...unselected}
        />
      ).toJSON()

      expect(component).toMatchSnapshot()
    })
  })

  describe("buttonPress()", () => {
    it("should invoke setState and onPress props", () => {
      const onPress = jest.fn()

      const component = shallow(
        <ButtonToggle {...props} onPress={onPress} />)
      const instance = component.instance()
      const setState = jest.spyOn(instance, "setState")
      const touchable: any = component.find("TouchableOpacity").first().props()
      touchable.onPress()
      expect(onPress).toHaveBeenCalled()
      expect(setState).toHaveBeenCalled()
    })
  })
})
