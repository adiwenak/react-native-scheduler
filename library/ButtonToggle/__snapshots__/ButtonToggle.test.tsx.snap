// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScheduleView /> render() should render with selected state 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 5,
      "justifyContent": "center",
      "marginHorizontal": 2,
      "opacity": 1,
    }
  }
  testID="Button-Toggle"
>
  <Text
    style={
      Array [
        Object {
          "color": "#455A64",
          "fontSize": 12,
          "marginHorizontal": 7,
          "marginVertical": 5,
        },
        Object {
          "color": "#FAFAFA",
        },
      ]
    }
  >
    button1
  </Text>
</View>
`;

exports[`<ScheduleView /> render() should render with unselected state 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EEEEEE",
      "borderRadius": 5,
      "justifyContent": "center",
      "marginHorizontal": 2,
      "opacity": 1,
    }
  }
  testID="Button-Toggle"
>
  <Text
    style={
      Array [
        Object {
          "color": "#455A64",
          "fontSize": 12,
          "marginHorizontal": 7,
          "marginVertical": 5,
        },
      ]
    }
  >
    button1
  </Text>
</View>
`;
